packages/runner-sdk/lib/sync.ts

```diff
@@ -55,4 +55,15 @@ export abstract class NangoSyncBase extends NangoActionBase {
 
         return hasErrors;
     }
+
+    protected removeMetadata<T extends object>(results: T[]) {
+        return results.map((result) => {
+            if ('_nango_metadata' in result) {
+                const { _nango_metadata, ...rest } = result;
+                return rest;
+            }
+
+            return result;
```

Comment:
It's not possible with `delete`? that might help to not reassign a new object

Reply:
I think it is possible. When I read the delete code path we apply the records to the upsert, along with the delete, so there's a weird possibility of updating a record when soft deleting.

Reply:
Not sure I get what you mean. I was just saying doing:
```ts
return results.map((result) => {
            delete result._nango_metadata;
            return result;
});
```

Reply:
Oh, gotcha. I was thinking we had an eslint rule against that.

I was just saying that deleteBatch actually updates records as well as soft deleting, so we still have to filter it out.

Reply:
See 2e20deeca